azext_edge/edge/providers/rpsaas/adr/assets.py

```diff
@@ -412,6 +495,80 @@ def _build_default_configuration(
     return json.dumps(defaults)
 
 
+def _build_ordered_csv_conversion_map(sub_point_type: str, portal_friendly: bool = False) -> Dict[str, str]:
+    """Results in an ordered dict for headers"""
+    from collections import OrderedDict
+    csv_conversion_map = [
+        ("queueSize", "QueueSize" if portal_friendly else "Queue Size"),
+        ("observabilityMode", "ObservabilityMode" if portal_friendly else "Observability Mode"),
+    ]
+    if not portal_friendly or sub_point_type == "dataPoints":
+        csv_conversion_map.append(("samplingInterval", "Sampling Interval Milliseconds"))
+    if not portal_friendly:
+        csv_conversion_map.append(("capabilityId", "Capability Id"))
+    if sub_point_type == "dataPoints":
+        csv_conversion_map.insert(0, ("dataSource", "NodeID" if portal_friendly else "Data Source"))
+        csv_conversion_map.insert(1, ("name", "TagName" if portal_friendly else "Name"))
+    else:
+        csv_conversion_map.insert(0, ("eventNotifier", "EventNotifier" if portal_friendly else "Event Notifier"))
+        csv_conversion_map.insert(1, ("name", "EventName" if portal_friendly else "Name"))
+
+    # datasource, name, queuesize, observabilitymode, samplinginterval, capabilityid
+    return OrderedDict(csv_conversion_map)
+
+
+def _convert_sub_points_from_csv(sub_points: List[Dict[str, str]]):
+    csv_conversion_map = {
```

Comment:
Also if these are all constant value, should we put it in common or something?

Reply:
This is too specific to put in common.py. Want to keep it here so a reader can see the mess easily (especially since it feels like DOE terminology isn't final)